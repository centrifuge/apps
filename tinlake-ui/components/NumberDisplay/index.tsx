import { Decimal } from 'decimal.js-light'
import * as React from 'react'
import { addThousandsSeparators } from '../../utils/addThousandsSeparators'

interface Props {
  value: string
  precision?: number
  prefix?: string
  suffix?: string
  render?: (formatted: string, prefix: string | undefined, suffix: string | undefined) => React.ReactElement
}

const NumberDisplay: React.FC<Props> = ({ value, precision, prefix, suffix, render }: Props) => {
  const valueToDecimal = new Decimal(value.toString()).toFixed(precision)
  const formatted = addThousandsSeparators(valueToDecimal.toString())

  if (render) {
    return render(formatted, prefix, suffix)
  }

  return (
    <span>
      {prefix}
      {formatted}
      {suffix}
    </span>
  )
}

NumberDisplay.defaultProps = {
  value: '0',
  precision: 2,
  prefix: '',
  suffix: '',
}

export default NumberDisplay
