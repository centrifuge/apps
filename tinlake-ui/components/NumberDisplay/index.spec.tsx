import React from 'react'
import renderer from 'react-test-renderer'
import NumberDisplay from '.'

it('renders correctly', () => {
  const tree = renderer.create(<NumberDisplay value="134144" precision={3} prefix="o" suffix=" CCC" />).toJSON()
  expect(tree).toMatchSnapshot()
})
