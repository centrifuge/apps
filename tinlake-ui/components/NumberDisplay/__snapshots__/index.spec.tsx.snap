// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<span>
  o
  134,144.000
   CCC
</span>
`;
